packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -83,7 +85,7 @@ ComponentNodeManager.create = function ComponentNodeManager_create(renderNode, e
     if (isAngleBracket) {
       assert(`You cannot invoke the '${tagName}' component with angle brackets, because it's a subclass of Component. Please upgrade to GlimmerComponent. Alternatively, you can invoke as '{{${tagName}}}'.`, component.isGlimmerComponent);
     } else {
-      assert(`You cannot invoke the '${tagName}' component with curly braces, because it's a subclass of GlimmerComponent. Please invoke it as '<${tagName}>' instead.`, !component.isGlimmerComponent);
+      assert(`You cannot invoke the '${tagName}' component with curly braces, because it's a subclass of GlimmerComponent. Please invoke it as '<${tagName}>' instead.`, !component.isGlimmerComponent || isClosureComponent || isComponentHelper);
```

Comment:
Is `isClosureComponent` threaded through here just for the assertion?

Perhaps we can just move the assertion further up the stack and avoid this. It kind of stinks to be passing these flags around.